virt-v2v/cold/entrypoint.go

```diff
@@ -0,0 +1,236 @@
+				fmt.Println("Error creating ca cert link ", err)
+				os.Exit(1)
+			}
+		}
+		virtV2vArgs = append(virtV2vArgs, "-ip", "/etc/secret/secretKey")
+
+		if info, err := os.Stat(VDDK); err == nil && info.IsDir() {
+			virtV2vArgs = append(virtV2vArgs,
+				"-it", "vddk",
+				"-io", fmt.Sprintf("vddk-libdir=%s", VDDK),
+				"-io", fmt.Sprintf("vddk-thumbprint=%s", os.Getenv("V2V_fingerprint")),
+			)
+		}
+		virtV2vArgs = append(virtV2vArgs, "--", os.Getenv("V2V_vmName"))
+	}
+
+	if err := executeVirtV2v(virtV2vArgs); err != nil {
+		fmt.Println("Error executing virt-v2v command ", err)
+		os.Exit(1)
+	}
+
+	if source == OVA {
+		var err error
+		xmlFilePath, err = getXMLFile(DIR, "xml")
+		if err != nil {
+			fmt.Println("Error gettin XML file:", err)
+			os.Exit(1)
+		}
+
+		http.HandleFunc("/xml", xmlHandler)
+		http.HandleFunc("/shutdown", shutdownHandler)
+		server = &http.Server{Addr: ":8080"}
+
+		fmt.Println("Starting server on :8080")
+		if err := server.ListenAndServe(); err != http.ErrServerClosed {
+			fmt.Printf("Error starting server: %v\n", err)
+			os.Exit(1)
+		}
+	}
+}
+
+func checkEnvVariablesSet(envVars ...string) bool {
+	for _, v := range envVars {
+		if os.Getenv(v) == "" {
+			return false
+		}
+	}
+	return true
+}
+
+func genName(diskNum int) string {
+	if diskNum <= 0 {
+		return ""
+	}
+
+	letters := []int32("abcdefghijklmnopqrstuvwxyz")
+	index := (diskNum - 1) % len(letters)
+	cycels := (diskNum - 1) % len(letters)
+
+	return genName(cycels) + string(letters[index])
+}
+
+func LinkDisks(diskKind string, num int) (err error) {
+	disks, err := filepath.Glob(diskKind)
+	if err != nil {
+		fmt.Println("Error getting disks ", err)
+		return
+	}
+
+	for _, disk := range disks {
+		diskNum, err := strconv.Atoi(disk[num:])
+		if err != nil {
+			fmt.Println("Error geting disks names ", err)
+			return err
+		}
+		diskLink := fmt.Sprintf("%s/%s-sd%s", DIR, os.Getenv("V2V_vmName"), genName(diskNum+1))
+		diskImgPath := disk
+		if diskKind == FS {
+			diskImgPath = fmt.Sprintf("%s/disk.img", disk)
+		}
+		if err = os.Symlink(diskImgPath, diskLink); err != nil {
+			fmt.Println("Error creating disk link ", err)
+			return err
+		}
+	}
+	return
+}
+
+func executeVirtV2v(args []string) (err error) {
+	virtV2vCmd := exec.Command(args[0], args[1:]...)
+	virtV2vCmd.Stdout = os.Stdout
+	virtV2vCmd.Stderr = os.Stderr
+
+	fmt.Println("exec ", virtV2vCmd)
+	if err = virtV2vCmd.Start(); err != nil {
+		fmt.Printf("Error executing command: %v\n", err)
+		return
+	}
+
+	virtV2vMonitorCmd := exec.Command("/usr/local/bin/virt-v2v-monitor")
+	virtV2vMonitorCmd.Stdin, _ = virtV2vCmd.StdoutPipe()
+	virtV2vMonitorCmd.Stdout = os.Stdout
+	virtV2vMonitorCmd.Stderr = os.Stderr
+
+	if err = virtV2vMonitorCmd.Start(); err != nil {
+		fmt.Printf("Error executing monitor command: %v\n", err)
+		return
+	}
+
+	if err = virtV2vCmd.Wait(); err != nil {
+		fmt.Printf("Error waiting for virt-v2v to finish : %v\n", err)
+		return
+	}
+	return
```

Comment:
how about trying this:
```
virtV2vCmd := exec.Command(args[0], args[1:]...)
virtV2vCmd.Stdout = os.Stdout
virtV2vCmd.Stderr = os.Stderr

virtV2vMonitorCmd := exec.Command("/usr/local/bin/virt-v2v-monitor")
virtV2vMonitorCmd.Stdin, _ = virtV2vCmd.StdoutPipe()
virtV2vMonitorCmd.Stdout = os.Stdout
virtV2vMonitorCmd.Stderr = os.Stderr

if err = virtV2vMonitorCmd.Start(); err != nil {
	fmt.Printf("Error executing monitor command: %v\n", err)
	return
}

if err = virtV2vCmd.Run(); err != nil {
	fmt.Printf("Error executing command: %v\n", err)
	return
}

if err = virtV2vCmd.Wait(); err != nil {
	fmt.Printf("Error waiting for virt-v2v to finish : %v\n", err)
	return
}
return
```

Reply:
and if we really wish to mimic the bash command we had, maybe we need to set 
virtV2vCmd.Stderr = os.Stdout
as `|&` is setting the stderr of virt-v2v to stdout and then passing the stdout to stdin of the monitor command

Reply:
the problem is that virt-v2v's stdout is consumed by the terminal so it can't be consumed by the monitor. It has to be either sent to both using `tee` or displayed by the monitor

Reply:
yeah, it didn't really worked in the previous implementation as well, and in addition the command missed args. I'm trying to redirect the output now with `TeeReader` following @bennyz suggestion.